### PR TITLE
RATIS-1589. Get ratis-shell resource from bin tarball

### PR DESCRIPTION
**Pull request by codings-dan**
see https://issues.apache.org/jira/projects/RATIS/issues/RATIS-1589?filter=allissues

### CODE REVIEW

**Comment by codings-dan** [comment]
@szetszwo Can you help review this pull request, thank you!

File: ratis-docs/content/cli.md [hunk 1]
```diff
@@ -41,12 +41,22 @@ $ mvn -DskipTests -Prelease -Papache-release clean package assembly:single
...
 Extract the ratis-shell tarball.
 ```
-$ tar -C <DST_DIR>/ratis-shell -xzf ratis-assembly/target/apache-ratis-<VERSION>-shell.tar.gz
+$ mkdir <DST_DIR>/ratis-shell
+$ tar -C <DST_DIR>/ratis-shell -xzf ratis-assembly/target/apache-ratis-<VERSION>-shell.tar.gz --strip-component 1
+```
+
+You can get the ratis-shell from bin tarball as well: 
+Download the Ratis bin tarball from https://ratis.apache.org/downloads.html .
+Note that ratis-shell is available starting from version 2.3.0
```
[the remark this answers is not in the document]
Comment (codings-dan) [reply ratis-docs/content/cli.md]: done

File: ratis-docs/content/cli.md [hunk 2]
```diff
@@ -41,12 +41,22 @@ $ mvn -DskipTests -Prelease -Papache-release clean package assembly:single
...
+Extract the bin tarball to a destination directory `<DST_DIR>`
+```
+$ tar -C <DST_DIR> -zxvf apache-ratis-<VERSION>-bin.tar.gz
```
[the remark this answers is not in the document]
Comment (codings-dan) [reply ratis-docs/content/cli.md]: Good idea, I have change it.

**Comment by codings-dan** [comment]
@szetszwo PTAL again, thx!

File: ratis-docs/content/cli.md [hunk 3]
```diff
@@ -41,12 +42,24 @@ $ mvn -DskipTests -Prelease -Papache-release clean package assembly:single
...
 Extract the ratis-shell tarball.
 ```
-$ tar -C <DST_DIR>/ratis-shell -xzf ratis-assembly/target/apache-ratis-<VERSION>-shell.tar.gz
+$ mkdir <DST_DIR>/ratis-shell
+$ tar -C <DST_DIR>/ratis-shell -xzf ratis-assembly/target/apache-ratis-<VERSION>-shell.tar.gz --strip-component 1
+```
+
+### Setting up from a binary tarball
+Download the Ratis bin tarball from https://ratis.apache.org/downloads.html .
+Note that ratis-shell is available starting from version 2.3.0 (already released) does not have ratis-shell bin.
```
Comment (szetszwo) [inline ratis-docs/content/cli.md:51]: This sentence "... available ... does not have ..." is a little bit confusing.   How about we rewrite it as below?
```
Note that the bin tarball of Ratis version 2.3.0 or earlier does not contain ratis-shell.
The bin tarball of later versions will contain ratis-shell.
```

**Review by szetszwo — COMMENTED** [review]
@codings-dan , thanks for the update.  Just a comment inlined.

Also, let's add a table right before "Setting up the ratis-shell".
```
## Availability
| Version  | Available in src tarball? | Available in bin tarball? |
| :------: | :-----------------------: | :-----------------------: |
| < 2.3.0  | No                        | No                        |
| 2.3.0    | Yes                       | No                        |
| \> 2.3.0 | Yes                       | Yes                       |

## Setting up the ratis-shell
```

**Comment by codings-dan** [comment]
> Also, let's add a table right before "Setting up the ratis-shell".

Adding a table is great, now it's all clear.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by codings-dan** [comment]
Thanks @szetszwo help review the pull request!

**Pull request merged**